### PR TITLE
Use platform-independent path separator

### LINKED ISSUE #116: Android classes from Android SDK in pitest classpath with compile compatibility set to `JavaVersion.VERSION_11`
## Describe the bug
* Minimal Android App setup with gradle-pitest-plugin
* Set compile compatibility to JavaVersion.VERSION_11 like
```
android {
   compileOptions {
     sourceCompatibility = JavaVersion.VERSION_11
     targetCompatibility = JavaVersion.VERSION_11
   }
}
```
* disable mockable Android jar `excludeMockableAndroidJar.set(true)` because Android classes come into the project by using alternatives like UnMock or Robolectric
* Runninng pitest task results in errors because android.jar from Android SDK is added to the classpath of pitest and so the Android classes are loaded from there (instead of using the intended ones from e.g. UnMock or Robolectric, custom shadows etc.)

Root cause is that android.jar from Android SDK is always added to classpath when executing Pitest (can be seen in verbose output). This has no effect when excludeMockableAndroidJar is disabled as the mockable Android jar is always added at the top of the classpath and therefore "wins" classloading. But when excludeMockableAndroidJar is enabled it loads the Android classes from the android.jar instead of from alternative source like UnMock etc.

## Expected behavior
No android.jar is added to pitest classpath

## Version used
pl.droidsonroids.pitest 0.2.15

## Suggested Solution
It seems that this is a fix of the Android Gradle Plugin to fix issues with the Java module system introduces with Java 9 (see [this](https://issuetracker.google.com/issues/168044181#comment3) and [this](https://issuetracker.google.com/issues/139013660) issue in Android issue tracker).
So android.jar is probably added to classpath via [PitestPlugin#241](https://github.com/koral--/gradle-pitest-plugin/blob/master/src/main/groovy/pl/droidsonroids/gradle/pitest/PitestPlugin.groovy#L241) and [PitestPlugin#245](https://github.com/koral--/gradle-pitest-plugin/blob/master/src/main/groovy/pl/droidsonroids/gradle/pitest/PitestPlugin.groovy#L245), thus it seems android.jar is not added via a configuration but directly to the compileTask classpath. 

Thus, Android.jar should be somehow removed from the classpath.

### PR DESCRIPTION
**Pull request by koral--**
Fixes #116

### CODE REVIEW

File: src/main/groovy/pl/droidsonroids/gradle/pitest/PitestPlugin.groovy [hunk 1]
```diff
@@ -67,7 +67,7 @@ class PitestPlugin implements Plugin<Project> {
     private static final String PITEST_JUNIT5_PLUGIN_NAME = "junit5"
     private final static List<String> DYNAMIC_LIBRARY_EXTENSIONS = ['so', 'dll', 'dylib']
     private final static List<String> DEFAULT_FILE_EXTENSIONS_TO_FILTER_FROM_CLASSPATH = ['pom'] + DYNAMIC_LIBRARY_EXTENSIONS
-    private final static Pattern MOCKABLE_ANDROID_JAR_PATH_PATTERN = Pattern.compile('.*/platforms/android-.*/android.jar$')
+    private final static Pattern MOCKABLE_ANDROID_JAR_PATH_PATTERN = Pattern.compile(".*\\${File.separator}platforms${File.separator}android-.*${File.separator}android.jar\$")
```
Comment (szpak) [inline src/main/groovy/pl/droidsonroids/gradle/pitest/PitestPlugin.groovy:70]: It doesn't look good:
```suggestion
    private final static Pattern MOCKABLE_ANDROID_JAR_PATH_PATTERN = Pattern.compile(".*${File.separator}platforms${File.separator}android-.*${File.separator}android.jar\$")
```
 ?

Comment (szpak) [reply src/main/groovy/pl/droidsonroids/gradle/pitest/PitestPlugin.groovy:70]: https://gwc-experiment.appspot.com/?g=groovy_3_0&codez=eNptjEEKwjAQRfc9RSiFpoKTE7h17QHcjM1UU5JOmExFEO9uF1UK-lcfHu95Gkw2BzPiHWHWEEHoSg84oSrJBD2nHCLZGnbN87g8KJRRUFleOaIOLKn8EJy8cPD7P9KKYEQ5N3VXVVgKiZoMCbW_kdh2YHbftvu0LihuI7fdKhS7RMxmb7WWSRY

Comment (szpak) [reply src/main/groovy/pl/droidsonroids/gradle/pitest/PitestPlugin.groovy]: > and it failed with:

Sure, on windows is `\p` which could be treated as a "control sequence". On Linux is not a problem.

Comment (koral--) [reply src/main/groovy/pl/droidsonroids/gradle/pitest/PitestPlugin.groovy]: It seems that even Groovy standard library itself does not support Windows 😅 
```
    !project.tasks.getByName('pitestRelease').additionalClasspath.getAsPath().contains('android.jar')
     |       |     |                          |                   |
     |       |     task ':pitestRelease'      file collection     java.util.regex.PatternSyntaxException: Unexpected internal error near index 1
```
I've tried several ways to examine a FileCollection but all them fail with the same exception on Appveyor Windows machine only.

**Comment by Pfoerd** [comment]
🚀

File: src/test/groovy/pl/droidsonroids/gradle/pitest/PitestPluginTest.groovy [hunk 2]
```diff
@@ -46,6 +47,7 @@ class PitestPluginTest extends Specification {
 
     @Issue("https://github.com/koral--/gradle-pitest-plugin/issues/116")
     void "excludes mockable Android JAR"() {
+        Assume.assumeFalse(System.getProperty("os.name", "unknown").toLowerCase(Locale.ROOT).contains("win"))
```
Comment (szpak) [inline src/test/groovy/pl/droidsonroids/gradle/pitest/PitestPluginTest.groovy:50]: Btw, [`@Ignoreif`](https://spockframework.org/spock/javadoc/2.2/spock/lang/IgnoreIf.html) could be even more readable.

(and I see that we should to add an example to javadoc for that extension :) ).

Comment (szpak) [reply src/test/groovy/pl/droidsonroids/gradle/pitest/PitestPluginTest.groovy:50]: Btw2, my old blog post - https://solidsoft.wordpress.com/2014/09/17/whats-new-in-upcoming-spock-1-0-simpler-conditional-test-execution-with-requires-and-ignoreif/

```
@IgnoreIf({ os.windows  }) 
```

**Pull request merged**